### ISSUE TITLE
Suggestion - Integrate MobileSAM into the pipeline for lightweight and faster inference

### ISSUE THREAD

**Issue by mdimtiazh** [CASIA-IVA-Lab/FastSAM#44]
Reference: https://github.com/ChaoningZhang/MobileSAM

 

Our project performs on par with the original SAM and keeps exactly the same pipeline as the original SAM except for a change on the image encode, therefore, it is easy to Integrate into any project.

 

MobileSAM is around 60 times smaller and around 50 times faster than original SAM, and it is around 7 times smaller and around 5 times faster than the concurrent FastSAM. The comparison of the whole pipeline is summarzed as follows:

 

![image](https://github.com/IDEA-Research/Grounded-Segment-Anything/assets/102718175/ecc801de-a7c9-4ccb-84f7-477e359a1892)

 

![image](https://github.com/IDEA-Research/Grounded-Segment-Anything/assets/102718175/39f257e2-dca0-4956-bdb1-4326e6b5d13b)

 

 

 

Best Wishes,
Qiao

**Comment by YinglongDu** [comment]
Thanks for your attention.

**Comment by ryouchinsa** [comment]
Here is the [Segment Anything CPP Wrapper for macOS](https://github.com/ryouchinsa/sam-cpp-macos/tree/master).
This code is originated from [Segment Anything CPP Wrapper](https://github.com/dinglufe/segment-anything-cpp-wrapper) and implemented on macOS app [RectLabel](https://rectlabel.com). We customized the original code so that getMask() uses the previous mask result called as low_res_logits and retain the previous mask array for undo/redo actions. 

We hope this macOS version would help to develop the iOS version.
Please let us know your opinion.

![sam_polygon](https://github.com/ryouchinsa/sam-cpp-macos/assets/1954306/4640e139-c533-4b8c-b27b-e02a401b9bbd)
